GitLab CI lint not working anymore without authentication

GitLab CI linting PR check fails (see details below)

Possible actions:

* Remove GitLab ci linting step
* Keep the GitLab CI linting step, but use authentication
  * Use git.vshn.net with a token
  * Use gitlab.com with a token
  * However, this check cannot be done in PR checks, since GitHub won't expose Secrets in PRs for security reasons (leaking etc.)

## Steps to Reproduce the Problem

1. `make test-cloudscale`

## Actual Behavior
See logs in https://github.com/appuio/component-openshift4-terraform/runs/2802667905?check_suite_focus=true, e.g.
```console

cat compiled/openshift4-terraform/openshift4-terraform/gitlab-ci.yml | docker run --rm -u "$(id -u)" -w /openshift4-terraform -v "${PWD}/compiled/openshift4-terraform/openshift4-terraform:/openshift4-terraform" -i docker.io/gableroux/gitlab-ci-lint:latest
...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  1957  100    30  100  1927    241  15540 --:--:-- --:--:-- --:--:-- 15782
configuration is invalid, unknown status 'null'
make: *** [Makefile:58: .test] Error 254
```

Reverse-engineering https://www.github.com/GabLeRoux/docker-gitlab-ci-lint showed that the Linting is done with the live API on gitlab.com. The request fails silently with HTTP status 400 from now on.

Turns out GitLab has put the CI-lint API behind authentication: https://about.gitlab.com/releases/2021/06/01/security-release-gitlab-13-12-2-released/#unauthenticated-ci-lint-api-may-lead-to-information-disclosure-and-ssrf

## Action points

1. Introduce a simple yaml lint for testing locally
1. Create a restricted token on git.vshn.net
2. Create GH secret so that the check works when linting with git.vshn.net API (set GITLAB_API_URL with user:token@https://... in workflow)
3. Separate the live CI linting from local make targets

--- a/Makefile
+++ b/Makefile
@@ -55,7 +55,7 @@ test-exoscale: .test ## Run tests for exoscale provider
 	rm compiled/$(COMPONENT_NAME)/$(COMPONENT_NAME)/backend.tf.json # either this, or make backend configurable
 	$(TERRAFORM_CMD) gitlab-terraform init
 	$(TERRAFORM_CMD) gitlab-terraform validate
-	$(GITLABCI_LINT_CMD)
+	$(YAMLLINT_DOCKER) -f parsable -c $(YAMLLINT_CONFIG) $(YAMLLINT_ARGS) -- $(compiled_path)/gitlab-ci.yml
 
 .PHONY: clean
 clean: ## Clean the project

--- a/Makefile.vars.mk
+++ b/Makefile.vars.mk
@@ -13,7 +13,7 @@ JSONNETFMT_ARGS ?= --in-place --pad-arrays
 JSONNET_IMAGE   ?= docker.io/bitnami/jsonnet:latest
 JSONNET_DOCKER  ?= $(DOCKER_CMD) $(DOCKER_ARGS) $(root_volume) --entrypoint=jsonnetfmt $(JSONNET_IMAGE)
 
-YAML_FILES      ?= $(shell find . -type f -not -path './vendor/*' \( -name '*.yaml' -or -name '*.yml' \))
+YAML_FILES      ?= $(shell find . -type f -not \( -path './vendor/*' -or -path '*.terraform/*' \) \( -name '*.yaml' -or -name '*.yml' \))
 YAMLLINT_ARGS   ?= --no-warnings
 YAMLLINT_CONFIG ?= .yamllint.yml
 YAMLLINT_IMAGE  ?= docker.io/cytopia/yamllint:latest
@@ -28,6 +28,3 @@ COMMODORE_CMD  ?= $(DOCKER_CMD) $(DOCKER_ARGS) $(root_volume) projectsyn/commodo
 
 TERRAFORM_IMAGE ?= $(shell grep image: class/defaults.yml | sed 's/ *//g' | cut -d : -f 2):$(shell grep tag: class/defaults.yml | sed 's/ *//g' | cut -d : -f 2)
 TERRAFORM_CMD   ?= $(DOCKER_CMD) $(DOCKER_ARGS) $(compiled_volume) $(extra_args) $(TERRAFORM_IMAGE)
-
-GITLABCI_LINT_IMAGE ?= docker.io/gableroux/gitlab-ci-lint:latest
-GITLABCI_LINT_CMD ?= cat $(compiled_path)/gitlab-ci.yml | $(DOCKER_CMD) $(DOCKER_ARGS) $(compiled_volume) -i $(GITLABCI_LINT_IMAGE)

